### PR TITLE
test: guppy -> pytket lowering with tket2

### PR DESCRIPTION
**Pull request by aborgna-q**
Adds the tests removed from tket2 in https://github.com/CQCL/tket2/pull/516

This requires a tket2 alpha release to avoid the git ref dependency.

### CODE REVIEW

File: pyproject.toml
Comment (aborgna-q) [inline pyproject.toml:45]: I will wait for the tket2 release before merging this

**Review by mark-koch — APPROVED** [review]
Looks good! Let's tame those cycles :D

File: tests/util.py
Comment (mark-koch) [inline tests/util.py:45]: That would be ideal, but I'm fine with leaving it here for this PR

File: tests/util.py
Comment (aborgna-q) [reply tests/util.py:12]: It seems to work here, but I had to `# type: ignore[import-not-found]` a similar case in tket2, so I'll add that here

**Pull request merged**
